Added support for getreceivedbyaddress.

polac opened the pull request.

polac committed:
Added support for getreceivedbyaddress.

jojkaart commented on src/fi/bittiraha/walletd/WalletRPC.java:
```diff
@@ -148,6 +149,18 @@ private String getnewaddress() {
 
+  private BigDecimal getreceivedbyaddress(String address) {
+    Coin retval = Coin.ZERO;
+	List<TransactionOutput> unspent = kit.wallet().calculateAllSpendCandidates(false, false);
```
calculateAllSpendCandidates returns a list of unspent outputs. You'll need to scan all wallet transactions to implement the equivalent of what bitcoind does.

polac committed:
Now summing up all outputs regardles if spent or not.

No direct access to wallet.transactions. So, had to use pools. Excluding
pools DEAD and PENDING because bitcoind has this optional minconf-parameter
that has default value 1. No support for that optional parameter but works
as if set to its default value.

jojkaart commented:
This looks almost perfect now. But it's still different from the bitcoind implementation. 
Basically two differences are still there:
1) There's no support for specifying the minconf parameter
2) bitcoind calculates the amount with minconf = 1 when it's not specified. This code does the same as minconf=0.

Could you fix these two differences? I can then merge this.

polac committed:
Added optional minconf-parameter.

polac commented:
1) added
2) It doesn't matter anymore but just wanted mention that my previous implementation was only accepting transactions which have minconf=1. I was only using pools SPENT and UNSPENT. Transaction can be in those pools only after it has at least 1 conf. Or have I misunderstood the way pools work. Anyways, doesn't matter anymore.

jojkaart commented:
This pull request has now been merged. Thank you for the contribution.
